PR feature/core_dump

favallone20 opened the pull request:
[DOC] Add documentation to explain how to use a core dump file 
-[MAK] Add images useful in the README file
 -[ADD] Add simple source code which divides by 0

Commit:
[DOC] Add documentation to explain how to use a core dump file
 -[MAK] Add images useful in the README file
 -[ADD] Add simple source code which divides by 0

LorenzoPagliara commented on core_dump/README.md:
```diff
@@ -0,0 +1,60 @@
+
+There are some situations where we would like to analyze a malfunction, but in the case of an air fighter in flight it is impossible to use the debbuger.
+
+In this case, a snapshot of the memory is useful to understand where is the bug in our code and avoid another time critical moment like this one.
```
Better "where the bug is in our code"

LorenzoPagliara commented on core_dump/README.md:
````diff
@@ -0,0 +1,60 @@
+ulimit -c
+```
+
+ulimit is a built-in Linux shell command that allows viewing or limiting system resource amounts that individual users consume. In particular, with "-c" this command returns the core dump size (expressed in the number of 512-byte blocks).
````
ulimit -> `ulimit`

LorenzoPagliara replied:
with "-c" ->with the `-c` option

LorenzoPagliara commented on core_dump/README.md:
````diff
@@ -0,0 +1,60 @@
+ulimit -c unlimited
+```
+
+In this project, a simple c program has been written where there a number is divided by 0.
````
Better "in which a number is divided by 0"

LorenzoPagliara commented on core_dump/README.md:
````diff
@@ -0,0 +1,60 @@
+./a.out
+```
+
+The -g option instructs the compiler to include debugging symbols in the compiled program, which enables a programmer to debug the program using a debugger like gdb.
````
-g -> `-g`

LorenzoPagliara commented on core_dump/README.md:
````diff
@@ -0,0 +1,60 @@
+If you prefer, you can change the directory changing the parameter kernel "kernel.core_pattern" through the following command:
+
+```bash
+sudo sysctl -w kernel.core_pattern=/var/crash/core-%e.%p.%h.%t
````
Which is the meaning of "%e.%p.%h.%t"?

LorenzoPagliara commented on core_dump/README.md:
```diff
@@ -0,0 +1,60 @@
+
+### How to use the core dump file
+
+Now, you are able to use the core dump file, with the gdb command which will jump to the moment captured by the core file, running the following instruction:
```
gdb -> `gdb`

When you see the highlighted command it means that it was inserted ` before and after the command

LorenzoPagliara commented on core_dump/README.md:
```diff
@@ -0,0 +1,60 @@
+
+The result is shown in the picture below.
+
+![gdb_with_core_dump](image/gdb_with_core.png)
```
Instead of inserting the picture, why don't you insert the output of the shell in a text block?

The text block is equal to the block used for the bash code, but you have to write "text" instead of "bash". Alternatively, you can also use "console".

```text
lorenzopagliara@katana-gf66-12ud-lorenzo:~/Personal/Projects/analysis_tools_for_programming/core_dump/src$ gdb a.out /var/lib/apport/coredump/core._home_lorenzopagliara_Personal_Projects_analysis_tools_for_programming_core_dump_src_a_out.1000.2280dfb0-4985-4849-9306-adbfea7a8f9b.9017.62996 
GNU gdb (Ubuntu 12.1-3ubuntu2) 12.1
Copyright (C) 2022 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from a.out...
[New LWP 9017]

This GDB supports auto-downloading debuginfo from the following URLs:
https://debuginfod.ubuntu.com 
Enable debuginfod for this session? (y or [n]) y
Debuginfod has been enabled.
To make this setting permanent, add 'set debuginfod enabled on' to .gdbinit.
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Core was generated by `./a.out'.
Program terminated with signal SIGFPE, Arithmetic exception.
#0  0x0000558a50909148 in main (argc=1, argv=0x7ffddbc95888) at main.c:11
11	    x = x/0;
(gdb) 

```

LorenzoPagliara commented on analysis_tools_for_programming.zip:
Can you delete this?

LorenzoPagliara commented on core_dump/src/main.c:
```diff
@@ -0,0 +1,14 @@
+#include<stdlib.h>
+#include <stdio.h>
+
+#define DIM 10
```
This is not used

Commit:
[MAK] Delete useless zip file
 -[DOC] Update readme with an explanation of sysctl command
 -[REF] Refactor code deleting unused define

Commit:
[DOC] Update Readme
